MekHQ/src/mekhq/campaign/mission/AtBScenario.java

```diff
@@ -492,6 +487,38 @@ public void setMapFile() {
         setMapFile(getTerrainType());
     }
 
+    /**
+     * If there are maps of the appropriate size available and we roll higher than
+     * the given threshold, replace the scenario's generated map with a fixed map
+     * from data/boards
+     */
+    public void setScenarioMap(int mapChance) {
```

Comment:
So something I've been thinking about: when we check for maps of `x` size is there value in adding some flexibility? For example: `<=x*1.1 && >=x*0.9`, this would increase the chances of hitting a valid map.

Reply:
xVariance and yVariance increases the range of board sizes searched from the current size.   making the variance a campaign option would be nice to have.
```
    public List<String> getMatchingBoards(int width, int height, int xVariance, int yVariance,
            List<Tags> tags) {
```

Reply:
Ah ha! Completely missed that. I agree it would be nice to have as a campaign option.